components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -2292,6 +2255,43 @@ public Group updateGroup(Group oldGroup, Group newGroup, Map<String, Boolean> re
         return newGroup;
     }
 
+    /**
+     * Perform user validation, check provided added member(s) details are exists in the user store. Else throw
+     * corresponding error
+     *
+     * @param userName
+     * @param userStoreDomainForGroup
+     * @param displayName
+     * @param addedUserIdsList
+     * @throws IdentitySCIMException
+     * @throws org.wso2.carbon.user.core.UserStoreException
+     */
+    private void doUserValidation(String userName, String userStoreDomainForGroup, String displayName,
+            List<Object> addedUserIdsList) throws IdentitySCIMException, org.wso2.carbon.user.core.UserStoreException {
+
+        // Compare user store domain of group and user store domain of user name, if there is a mismatch do not
+        // update the group.
+        String userStoreDomainForUser = IdentityUtil.extractDomainFromName(userName);
+        if (!isInternalOrApplicationGroup(userStoreDomainForGroup) && !userStoreDomainForGroup
+                .equalsIgnoreCase(userStoreDomainForUser)) {
+            throw new IdentitySCIMException(userName + " does not belongs to user store " + userStoreDomainForGroup);
+        }
+
+        // Check if the user ids & associated user name sent in updated (new) group exist in the user store.
+        if (userName != null) {
+            String userId = carbonUM.getUserClaimValue(userName, SCIMConstants.CommonSchemaConstants.ID_URI, null);
+            if (userId == null || userId.isEmpty()) {
+                String error = "User: " + userName + " doesn't exist in the user store. Hence, can not update the "
+                        + "group: " + displayName;
+                throw new IdentitySCIMException(error);
+            } else {
+                if (!UserCoreUtil.isContain(userId, addedUserIdsList.toArray(new String[addedUserIdsList.size()]))) {
+                    throw new IdentitySCIMException("Given SCIM user Id and name not matching..");
```

Comment:
A grammatical error in the error message.

Reply:
Fixed